packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts

```diff
@@ -88,6 +89,24 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
     });
   }
 
+  /**
+   * Add a redirection listener to this load balancer
+   */
+  public addRedirect(props?: ApplicationLoadBalancerRedirectConfig): ApplicationListener {
```

Comment:
This is the preferred convention for cases like this; it will also simplify some of the `props?.foo` accessors below.

```suggestion
  public addRedirect(props: ApplicationLoadBalancerRedirectConfig = {}): ApplicationListener {
```